src/user.ts

```diff
@@ -120,6 +120,21 @@ export class User {
         await this.storage.username;
     }
 
+    async usernameExists(username: string): Promise<boolean> {
+        const normalizedUsername = username.normalize('NFKC');
+
+        const fullUsername = TonomyUsername.fromUsername(
+            normalizedUsername,
+            AccountType.PERSON,
+            getSettings().accountSuffix
+        );
+
+        const user: API.v1.AccountObject = (await User.getAccountInfo(fullUsername)) as any; // Throws error if username is taken
+
+        if (user) return true;
```

Comment:
because L132 throws if username taken, you don't need to check for user here, could just return true

As this test return a boolean, better to catch the error of User.getAccountInfo(), make sure it is the right type of error and then return true or false properly